scipy/stats/_continuous_distns.py

```diff
@@ -832,8 +832,29 @@ def fit(self, data, *args, **kwds):
         return a, b, floc, fscale
 
     def _entropy(self, a, b):
-        return (sc.betaln(a, b) - (a - 1) * sc.psi(a) -
-                (b - 1) * sc.psi(b) + (a + b - 2) * sc.psi(a + b))
+        def regular(a, b):
+            return (sc.betaln(a, b) - (a - 1) * sc.psi(a) -
+                    (b - 1) * sc.psi(b) + (a + b - 2) * sc.psi(a + b))
+
+        def asymptotic_ab_large(a, b):
+            sum_ab = a + b
+            log_term = 0.5 * (
+                np.log(2 * np.pi * a * b) - 3 * np.log(sum_ab) + 1
+            )
+            t1 = (
+                (110 * sum_ab**3 + 20 * sum_ab**2 + sum_ab - 2)
+                / (120 * sum_ab**4)
+            )
+            t2 = (1 - 50 * a**3 - 10 * a**2 - a) / (120 * a**4)
+            t3 = (1 - 50 * b**3 - 10 * b**2 - b) / (120 * b**4)
+            return log_term + t1 + t2 + t3
+
+        return _lazywhere(
+            a >= 5e6 and b >= 5e6,
```

Comment:
5e6 seems to be too high. Please reduce the threshold to about 4.9e6 or so.
![Figure_1](https://github.com/scipy/scipy/assets/33975162/8a9bed13-9b04-4177-96a3-a97c3484f05f)